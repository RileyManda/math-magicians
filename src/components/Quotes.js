import React, { useState, useEffect } from 'react';
import apiKey from '../api/config';

function Quotes() {
  const [quotes, setQuotes] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const res = await fetch(
        'https://api.api-ninjas.com/v1/quotes?category=happiness',
        {
          headers: {
            'X-Api-Key': apiKey,
          },
        },
      );
      const json = await res.json();
      console.log(json);
      setQuotes(json);
    };
    fetchData();
  }, [setQuotes]);
  return (
    <div>

      {quotes.map((data) => (
        <p key={data.id}>{data.quote}</p>
      ))}

    </div>
  );
}
export default Quotes;
