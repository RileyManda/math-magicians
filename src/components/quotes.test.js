import React from 'react';
import { render, act } from '@testing-library/react';
import Quotes from './Quotes';

describe('Quotes', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });

  it('should render loading state', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });

  it('should render error state', async () => {
    const error = new Error('Failed to fetch data');
    jest.spyOn(global, 'fetch').mockRejectedValueOnce(error);

    await act(async () => {
      const { asFragment } = render(<Quotes />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
