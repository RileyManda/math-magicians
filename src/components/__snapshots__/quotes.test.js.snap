// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes should match the snapshot 1`] = `
<div>
  <div
    class="QuotesContainer"
  >
    <div
      class="Quotes"
    >
      <div>
        <p>
          Loading...
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Quotes should render error state 1`] = `<DocumentFragment />`;

exports[`Quotes should render loading state 1`] = `
<div>
  <div
    class="QuotesContainer"
  >
    <div
      class="Quotes"
    >
      <div>
        <p>
          Loading...
        </p>
      </div>
    </div>
  </div>
</div>
`;
